frappe/integrations/doctype/paypal_settings/paypal_settings.py

```diff
@@ -363,7 +363,27 @@ def manage_recurring_payment_profile_status(profile_id, action, args, url):
 	})
 
 	response = make_post_request(url, data=args)
+	
 
 	if response.get("ACK")[0] != "Success":
 		frappe.throw(_("Failed while amending subscription"))
 
+@frappe.whitelist(allow_guest=True)
+def ipn_handler():
+	data = frappe.local.form_dict
```

Comment:
there must be some validation before accepting records where `allow_guest=True` 

Otherwise someone can use this to do DoS